lib/gringotts/gateways/authorize_net.ex

```diff
@@ -386,38 +382,30 @@ defmodule Gringotts.Gateways.AuthorizeNet do
     HTTPoison.request(method, path, payload, headers)
   end
 
-  # Function to return a response 
+  # Function to return a response
   defp respond({:ok, %{body: body, status_code: 200}}) do
     raw_response  = naive_map(body)
-    cond do
-      raw_response[@response_type[:auth_response]] ->
-        response_check(raw_response[@response_type[:auth_response]], raw_response)
-      raw_response[@response_type[:transaction_response]] ->
-        response_check(raw_response[@response_type[:transaction_response]], raw_response)
-      raw_response[@response_type[:error_response]] ->
-        response_check(raw_response[@response_type[:error_response]], raw_response)
-      raw_response[@response_type[:customer_profile_response]] ->
-        response_check(raw_response[@response_type[:customer_profile_response]], raw_response)
-      raw_response[@response_type[:customer_payment_profile_response]] ->
-        response_check(raw_response[@response_type[:customer_payment_profile_response]], raw_response)
-      raw_response[@response_type[:delete_customer_profile]] ->
-        response_check(raw_response[@response_type[:delete_customer_profile]], raw_response)  
-    end
+    response_type = ResponseHandler.check_response_type(raw_response)
+    response_check(raw_response[response_type], raw_response)
   end
 
-  defp respond({:error, %{body: body, status_code: code}}) do
-    {:error, Response.error(raw: body, code: code)}
+  defp respond({:ok, %{body: body, status_code: code}}) do
+    {:error, Response.error(params: body, error_code: code)}
```

Comment:
`error_code` in respond method where `status_code` can be any code ?
cc: @ashish173

Reply:
This respond method pattern matches to a `network error`. So if due to some reason you are not even able to reach the server then there won't be any `status` received so error and error_code.